### PR TITLE
(Draft) Feat: Client Auth Trait Func in Upstream Nodes

### PR DESCRIPTION
**Pull request by coleFD**
Added trait function `ParseDownstreamMiningMessages::downstream_is_authorized` for Upstream nodes to implement client authorization services if needed

### CODE REVIEW

File: protocols/v2/roles-logic-sv2/src/handlers/mining.rs [hunk 1]
```diff
@@ -63,6 +63,12 @@ pub trait ParseDownstreamMiningMessages<
         match (message_type, payload).try_into() {
             Ok(Mining::OpenStandardMiningChannel(mut m)) => {
                 debug!("Received OpenStandardMiningChannel message");
+                // check user auth
+                if !Self::downstream_is_authorized(self_mutex.clone(), &m.user_identity)? {
+                    return Ok(SendTo::Respond(Mining::OpenMiningChannelError(
+                        OpenMiningChannelError::new_unknown_user(m.get_request_id_as_u32()),
```
Comment (darricksee) [inline protocols/v2/roles-logic-sv2/src/handlers/mining.rs:69]: Can you add some info! or error! logging for these Error cases?

File: roles/v2/pool/src/lib/mining_pool/message_handler.rs [hunk 2]
```diff
@@ -228,6 +235,12 @@ impl ParseDownstreamMiningMessages<(), NullDownstreamMiningSelector, NoRouting>
             .map_err(|e| roles_logic_sv2::Error::PoisonLock(e.to_string()))?;
         match (message_type, payload).try_into() {
             Ok(Mining::OpenStandardMiningChannel(mut m)) => {
+                // check user auth
+                if !Self::downstream_is_authorized(self_mutex.clone(), &m.user_identity)? {
+                    return Ok(SendTo::Respond(Mining::OpenMiningChannelError(
+                        OpenMiningChannelError::new_unknown_user(m.get_request_id_as_u32()),
```
Comment (darricksee) [inline roles/v2/pool/src/lib/mining_pool/message_handler.rs:241]: Logging here too pls.

**Review by darricksee — APPROVED** [review]
Looks good - just add some logging please.

**Comment by coleFD** [comment]
linked https://github.com/stratum-mining/stratum/pull/514

**Pull request closed without merging**